download.py

```diff
@@ -1,29 +1,47 @@
-import json
-import sys
+import boto3
 import cgi
-import os
 import datetime
+import json
+import os
+import sys
+import traceback
 
 from do_authentication import authenticate
 from do_http_get import do_get
 
+aws_session_args = {}
+
 ##############################################################################################################
 # First Step: Get the config data from config.json file
 ##############################################################################################################
 
 try:
-    config_file = open("config.json", "r")
+    # In case we run the script from somewhere outside our working directory...
+    base_path = os.path.dirname(os.path.abspath(__file__))
+    config_file = open(base_path + "/config.json", "r")
     config = json.load(config_file)
     config_file.close()
 except:
     sys.stderr.write("Error loading config.json file.\n")
     exit(1)
 
+# Front load some of the AWS creds values
+aws_profile           = None
+aws_access_key_id     = None
+aws_secret_access_key = None
+
 # The config.json file must contain the following data:
-username = config['icann.account.username']
-password = config['icann.account.password']
-authen_base_url = config['authentication.base.url']
-czds_base_url = config['czds.base.url']
+username              = config['icann.account.username']
+password              = config['icann.account.password']
+authen_base_url       = config['authentication.base.url']
```

Comment:
One possible interpretation of [PEP 20](https://www.python.org/dev/peps/pep-0020/), from the Python style guide is to prefer more verbose variable names over the use of acronyms and abbreviations. As [most code is read more often that it is written](https://www.goodreads.com/work/quotes/3779106-clean-code-a-handbook-of-agile-software-craftsmanship-robert-c-martin), semantic variable naming improves readability.

This really boils down to style preference and interpretation so YMMV.

Reply:
Are you talking about line 36?

Reply:
Yes. `authen_base_url`.